plugins/amqp/src/plugin.cpp

```diff
@@ -600,8 +623,22 @@ class rabbitmq_saver final : public plugin_saver {
       .mandatory = args_.mandatory,
       .immediate = args_.immediate,
     };
-    return [&ctrl, engine = std::make_shared<amqp_engine>(std::move(*engine)),
-            opts = std::move(opts)](chunk_ptr chunk) mutable {
+    auto hb = try_get<uint64_t>(config_, "heartbeat");
+    if (hb && *hb && **hb > 0) {
+      // If we are requesting heartbeats, we are also responsible to handle the
+      // heartbeats we get. If we have long gaps in interaction with the broker,
+      // we need to proactively check explicitly if there is something for us.
+      // We check 3 times per heartbeat interval, at most once per second.
+      auto interval = std::max((int64_t) 1,
+        detail::narrow_cast<int64_t>(std::floor(**hb/3)));
+      TENZIR_DEBUG("using heartbeat interval {}", interval);
```

Comment:
```suggestion
      auto interval = std::max(uint64_t{1}, **hb / 3);
      TENZIR_DEBUG("using heartbeat interval of {} seconds", interval);
```

This can be simplified if you construct a `uint64_t` for the `interval` assignment.

Reply:
That gave me weird compile issues, but I'll try again. Thanks!

Reply:
Hmm... But `std::chrono::seconds` wants an `int64_t`.